### ISSUE TITLE
safedb.php is vulnerable to SQL injections

### ISSUE THREAD

**Issue by tniessen** [gautamkrishnar/nothing-private#6]
`safedb.php` contains two SQL injection vulnerabilities:

```js
$query =  "SELECT * FROM browsertab WHERE fingerprint='".$_GET['finger']."'";
...
$query =  "INSERT INTO browsertab VALUES('".$_GET['finger']."','".$_GET['name']."')";
```

SQL injections allow attackers to manipulate queries and gain access to the system (even though the possibilities are very limited when it comes to SQLite). You should never use a DBMS with applications which might contain SQL injection vulnerabilities, so you might want to learn more about SQL first. [This thread](http://stackoverflow.com/questions/60174/how-can-i-prevent-sql-injection-in-php) might get you started.

**Comment by gautamkrishnar** [comment]
Yes, I know that. I got some attacks previously. Since this is not a task critical system, i don't care even if the entire db is deleted.

**Comment by gautamkrishnar** [comment]
Thanks for reporting... I will surely add that code... 👍

**Comment by naltun** [comment]
If I am not mistaken, you can just `sanitize` your queries using [PHP filters](https://secure.php.net/manual/en/intro.filter.php). If you want I can go ahead and do it.

**Comment by gautamkrishnar** [comment]
Yes @naltun... You can go ahead with this... I am pretty busy to work on this project :+1:

**Comment by electron-volt** [comment]
The best option is to use prepared statements. I can help with this if you want - or you can easily do it yourself. A great reference is https://phpdelusions.net/sql_injection

**Comment by gautamkrishnar** [comment]
@electron-volt Thanks for the suggestion 😄.  yes, I am a little busy right now. Can't work on it. Please go ahead if you can fix it...

**Comment by electron-volt** [comment]
Alright, I'll get it fixed. Awesome project by the way.

**Comment by gautamkrishnar** [comment]
Thank you!

**Comment by naltun** [comment]
@electron-volt Thanks for assisting @gautamkrishnar and for the link on SQL injection prevention. @gautamkrishnar I've been on holiday, so I'm away... Glad someone offered to help out. Indeed, awesome project.

**Comment by gautamkrishnar** [comment]
fixed in #19
